### PR TITLE
Fix permissions issues on main.queries and users.queries.

### PR DESCRIPTION
**Pull request by corentin-soriano**
- Use `$session->get('user-id')` instead of user input when only the user himself can perform this action.
- Add additional checks on administratives tasks.
- `initialize_user_password` removed because it is unused.

@tvnnn can you test this branch please?

### CODE REVIEW

**Comment by tvnnn** [comment]
Hi @corentin-soriano, it looks good to me, the fixes have resolved the vulnerabilities I reported.

**Review by nilsteampassnet — APPROVED** [review]

**Review by nilsteampassnet — APPROVED** [review]

**Review by nilsteampassnet — APPROVED** [review]

File: sources/main.queries.php [hunk 1]
```diff
@@ -425,26 +425,69 @@ function userHandler(string $post_type, array|null|string $dataReceived, array $
...
-                $dataReceived['body'],
-                (string) filter_var($dataReceived['subject'], FILTER_SANITIZE_FULL_SPECIAL_CHARS),
-                (array) filter_var_array(
-                    $dataReceived['pre_replace'],
-                    FILTER_SANITIZE_FULL_SPECIAL_CHARS
+            // Get info about user to send email
+            $data_user = DB::queryfirstrow(
+                'SELECT admin, gestionnaire, can_manage_all_users, isAdministratedByRole FROM ' . prefixTable('users') . '
+                WHERE email = %s',
+                $post_id
```
Comment (nilsteampassnet) [inline sources/main.queries.php:459]: $post_id ne semble pas initialisé à cette étape

File: sources/main.queries.php [hunk 2]
```diff
@@ -425,26 +425,69 @@ function userHandler(string $post_type, array|null|string $dataReceived, array $
...
+            );
+
+            // Only administrators and managers can send mails
+            if (
+                // Administrator user
+                (int) $session->get('user-admin') === 1
+                // Manager of basic/ro users in this role but don't allow promote user to admin or managers roles
+                || ((int) $session->get('user-manager') === 1
+                    && in_array($data_user['isAdministratedByRole'], $session->get('user-roles_array'))
+                    && (int) $post_is_admin !== 1 && (int) $data_user['admin'] !== 1
```
Comment (nilsteampassnet) [inline sources/main.queries.php:469]: De meme que $post_is_admin, $post_is_hr et $post_is_manager

File: sources/main.queries.php [hunk 3]
```diff
@@ -526,32 +626,22 @@ function keyHandler(string $post_type, /*php8 array|null|string */$dataReceived,
...
-                && empty($dataReceived['user_id']) === false) {
-                // Use id requested from user if he is admin.
-                $userId = $dataReceived['user_id'];
-            } else {
-                // Use id from session if user not admin or id not sent.
-                $userId = $session->get('user-id');
-            }
-
             // Handle the case where no PWD is provided (user reset his own encryption keys).
             if (empty($dataReceived['user_pwd']) && (int) $userId === $session->get('user-id')) {
```
Comment (nilsteampassnet) [inline sources/main.queries.php:646]: $userId is not initialized

File: sources/main.queries.php (hunk 1 above)
Comment (corentin-soriano) [reply sources/main.queries.php:459]: I forgot to replace `$post_id` with `$dataReceived['receipt']`

File: sources/main.queries.php (hunk 2 above)
Comment (corentin-soriano) [reply sources/main.queries.php:469]: Copy/paste error.
We must check if the query returns a result and if the user has the right to administer the account found.
I will correct this tomorrow.

File: sources/main.queries.php (hunk 3 above)
Comment (corentin-soriano) [reply sources/main.queries.php:646]: `$filtered_user` is expected.

**Pull request merged**